dali/kernels/imgproc/resample/bilinear_impl.cuh

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_RESAMPLE_BILINEAR_IMPL_CUH_
+#define DALI_KERNELS_IMGPROC_RESAMPLE_BILINEAR_IMPL_CUH_
+
+#include <cuda_runtime.h>
+#ifndef __CUDACC__
+#include <algorithm>
+#endif
+#include "dali/kernels/static_switch.h"
+#include "dali/kernels/common/convert.h"
+
+namespace dali {
+namespace kernels {
+
+namespace {
+
+/// @brief Implements horizontal resampling for a custom ROI
+/// @param x0 - start column, in output coordinates
+/// @param x1 - end column (exclusive), in output coordinates
+/// @param y0 - start row
+/// @param y1 - end row (exclusive)
+template <int static_channels = -1, typename Dst, typename Src>
+__device__ void LinearHorz_Channels(
+    int x0, int x1, int y0, int y1,
+    float src_x0, float scale,
+    Dst *out, int out_stride,
+    const Src *in, int in_stride, int in_w, int dynamic_channels) {
+  src_x0 += 0.5f * scale - 0.5f;
+
+  const int channels = static_channels < 0 ? dynamic_channels : static_channels;
+
+  for (int j = x0 + threadIdx.x; j < x1; j += blockDim.x) {
```

Comment:
Have you checked if loop unrolling may help with the perf?

Reply:
Not the outer loop. In general, I haven't measured performance - this has been more of an accuracy  problem.